compiler/semantic/scope.go

```diff
@@ -29,28 +30,28 @@ type entry struct {
 	order int
 }
 
-func (s *Scope) DefineVar(name string) error {
+func (s *Scope) DefineVar(id *ast.ID) error {
```

Comment:
Nit: Use a consistent name for the `*ast.ID` parameter to this, `DefineAs`, and `DefineConst`. I like `name` since it makes the parameter's role clear but `id` is fine too.